agent/agent.go

```diff
@@ -467,17 +467,25 @@ func (a *Agent) runPostGroupCommands(ctx context.Context, tc *taskContext) {
 	}
 }
 
-func (a *Agent) killProcs(tc *taskContext, ignoreTaskGroupCheck bool) {
+func (a *Agent) killProcs(tc *taskContext, ignoreTaskGroupCheck bool, loggerClosed bool) {
```

Comment:
There is (added recently) an isClosed field of the log harness we use. It's private right now, but what if we expose that rather than passing this in?

Reply:
I added a function to check `closed`.